nautobot/docs/development/apps/migration/migrating-an-app-from-v1.md

```diff
@@ -0,0 +1,191 @@
+        name="Logs",
+        permissions=["your_app.view_synclogentry"],
+    ),
+]
+
+menu_items = (
+    NavMenuTab(
+        name="Plugins",
+        groups=(NavMenuGroup(name="Your App", weight=1000, items=tuple(items)),),
+    ),
+)
+
+```
+
+### Replace DjangoFilterBackend with NautobotFilterBackend
+
+If your Custom `FilterBackend` class is derived from `DjangoFilterBackend`, you should replace `DjangoFilterBackend` with `NautobotFilterBackend`.
+
+### Revamp Rest API Serializers
+
+All `NestedSerializers` classes are removed from Nautobot v2.0. If any `NestedSerializers` exist for your models, you should just remove their class definitions and references. In addition, the `?brief=` API query parameter is replaced by `?depth=<0-10>`. As a result, the ability to specify `brief_mode` in `DynamicModelChoiceField`, `DynamicModelMultipleChoiceField`, and `MultiMatchModelMultipleChoiceField` has also been removed. For every occurrence of the aforementioned fields where you have `brief_mode` set to `True/False` (e.g. `brief_mode=True`), please remove the statement, leaving other occurrences of the fields where you do not have `brief_mode` specified as they are. Checkout our [API documentation](../../../user-guide/platform-functionality/rest-api/overview.md#depth-query-parameter) for this change.
+
+After removing existing `NestedSerializers`, you can change the `fields` attribute in your serializers' `class Meta` to `__all__` and that will automatically include all the model's fields in the serializer. If you want to exclude certain fields of the model, you can specify a list of fields you want to display in the `fields` attribute instead.
+
+Include all model attributes:
+
+```python
+class ExampleModelSerializer(NautobotModelSerializer):
+    """Used for normal CRUD operations."""
+
+    url = serializers.HyperlinkedIdentityField(view_name="plugins-api:example_plugin-api:anotherexamplemodel-detail")
+
+    class Meta:
+        model = AnotherExampleModel
+        fields = "__all__"
+```
+
+Include only specified model attributes:
+
+```python
+class ExampleModelSerializer(NautobotModelSerializer):
+    """Used for normal CRUD operations."""
+
+    url = serializers.HyperlinkedIdentityField(view_name="plugins-api:example_plugin-api:anotherexamplemodel-detail")
+
+    class Meta:
+        model = AnotherExampleModel
+        # example_attribute_4 is not included in the serializer
+        fields = ["url", "example_attribute_1", "example_attribute_2", "example_attribute_3"]
+```
+
+### Revamp CSV Import and Export
+
+CSV Import for models are now done automatically via the Rest API. As a result of this change, `CSVForms` classes are no longer needed and should be deleted. In addition, `csv_headers` and `to_csv` attributes should be removed from your model definition. Check out our [release notes](../../../release-notes/version-2.0.md#revamped-csv-import-and-export-2569-3715) for this specific change.
```

Comment:
from the "no longer needed and should be deleted", I understand that it's not mandatory to delete or not?

Reply:
They're no longer used in any way by core, so it won't *hurt* things if you don't delete them, but they'll be dead code.